enhancements/authentication/invalid-certs.md

```diff
@@ -0,0 +1,200 @@
+---
+title: invalid-certs
+authors:
+  - "@s-urbaniak"
+reviewers:
+  - "@sttts"
+  - "@stlaz"
+  - "@deads2k"
+approvers:
+  - "@mfojtik"
+creation-date: 2022-12-08
+last-updated: 2022-12-08
+---
+
+# Detecting invalid CA certificates
+
+## Summary
+
+OpenShift 4.10 is going to be rebased against Kubernetes 1.23. This requires using Go 1.17.
+However, starting with Go 1.17 support for invalid certificates is going to be removed, see https://go.dev/doc/go1.17.
+Formally, the temporary `GODEBUG=x509ignoreCN=0` flag has been removed.
+This implies that starting from OpenShift 4.10 invalid certificates will not be trusted any more as they will fail verification.
+
+Example:
+
+Given the following certificate:
+```
+Certificate:
+    Data:
+        ...
+        Subject: CN=foo-domain.com
+        X509v3 extensions:
+            X509v3 Key Usage: critical
+                Digital Signature, Key Encipherment
+            X509v3 Extended Key Usage: 
+                TLS Web Server Authentication
+            X509v3 Basic Constraints: critical
+                CA:FALSE
+```
+
+Verification against the `foo-domain.com` hostname of such certificate will fail with the following error in Go 1.17:
+```
+x509: certificate relies on legacy Common Name field, use SANs instead
+```
+
+Verification of server certificates is executed during TLS client handshakes,
+a TLS (https) client observing an invalid certificate will reject the connection attempt.
+
+Cluster internal issued certificates are not affected,
+however custom certificates can be configured in various cases:
+- custom serving certificates for kube-apiserver
+- custom API webhooks
+- custom aggregated API endpoints
+- custom certificates for route endpoints
+- certificates of external auth identity providers
```

Comment:
Shall we add certificates securing infrastructure API as a target case here?

For example, an OpenStack cloud exposing its API endpoints with a custom certificate. We have several platform-specific touchpoints (eg cluster-api-provider-openstack) that would need to expose the metric, if I understand this enhancement correctly

Reply:
i'd consider this as a separate enhancement proposal or separate bugzilla. This proposal was not meant to be catch-all to focus on scope for the auth team.

Reply:
OK. Opened https://bugzilla.redhat.com/show_bug.cgi?id=2038166 , thanks